### PR TITLE
Bump C# language version to 11

### PR DESCRIPTION
**Pull request by stephentoub**
### Motivation and Context

Use the latest C# language version.

### Description

Some places already had 11 set, others were still using 10.  This makes it consistently 11 everywhere.  C# 11 is currently the latest, with C# 12 in preview until November.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by MovGP0** [comment]
LGTM

File: dotnet/common.props
Comment (lemillermicrosoft) [inline dotnet/common.props:1]: Is `common.props` used by anything? I'm wondering if we can remove it.

File: dotnet/src/SemanticKernel.Skills/Skills.UnitTests/Skills.UnitTests.csproj
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Skills/Skills.UnitTests/Skills.UnitTests.csproj:1]: There's a handful of projects that don't specify any version, like `dotnet\src\SemanticKernel.IntegrationTests\SemanticKernel.IntegrationTests.csproj` -- should we add to those as well?

File: dotnet/common.props
Comment (stephentoub) [reply dotnet/common.props:1]: Doesn't look like it's referenced anywhere

File: dotnet/src/SemanticKernel.Skills/Skills.UnitTests/Skills.UnitTests.csproj
Comment (stephentoub) [reply dotnet/src/SemanticKernel.Skills/Skills.UnitTests/Skills.UnitTests.csproj:1]: > should we add to those as well?

It'd be better IMHO to have a single location where all of these common settings are picked up from.  Presumably that's the intent of the root Directory.props?

File: dotnet/common.props
Comment (lemillermicrosoft) [reply dotnet/common.props:1]: cc @shawncal might have more historical knowledge

File: dotnet/src/SemanticKernel.Skills/Skills.UnitTests/Skills.UnitTests.csproj
Comment (lemillermicrosoft) [reply dotnet/src/SemanticKernel.Skills/Skills.UnitTests/Skills.UnitTests.csproj:1]: Agreed. I'm not sure if there's a reason these UT projects and IntegrationTest, etc. don't reference the Directory.props -- and I'm again finding myself wondering why the SemanticKernel.csproj doesn't either -- does dotnet pick it up automatically? Either way, should we just remove the `<langversion> from the csproj all together? Also cc @shawncal

Comment (lemillermicrosoft) [reply dotnet/src/SemanticKernel.Skills/Skills.UnitTests/Skills.UnitTests.csproj:1]: https://learn.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2022#explore-customization-options

Found my own answer, so yeah, maybe we can just remove the `LangVersion` in the csprojs in this change.

**Review by lemillermicrosoft — DISMISSED** [review]

File: dotnet/common.props
Comment (shawncal) [reply dotnet/common.props:1]: I think you're right. I'll resolve the thread here, and fix it in an upcoming PR.

**Comment by shawncal** [comment]
@stephentoub Change looks good -- some files were moved in another change, and I cannot resolve the conflicts from here. Could you please sync up?

**Review by shawncal — APPROVED** [review]

**Pull request merged**